superagentx_handlers/websitecrawler/amazon.py

```diff
@@ -0,0 +1,190 @@
+                link = item.find(
+                    'a',
+                    {
+                        'class': 'a-link-normal s-underline-text s-underline-link-text s-link-style a-text-normal'
+                    }
+                ).get('href')
+                try:
+                    sale_prices = item.find_all('span', {'class': 'a-offscreen'})
+                    if sale_prices:
+                        sale_price = sale_prices[0].text.replace('$', '').replace(',', '').strip()
+                    else:
+                        sale_price = ''
+
+                    old_prices = item.find_all('span', {'class': 'a-offscreen'})
+                    if old_prices:
+                        old_price = old_prices[1].text.replace('$', '').replace(',', '').strip()
+                    else:
+                        old_price = ''
+                except:
+                    old_prices = item.find(
+                        'span',
+                        {
+                            'class': 'a-offscreen'
+                        }
+                    )
+                    old_price = float(
+                        old_prices.text.replace('£', '').replace(',', '').strip())
+                try:
+                    reviews = float(
+                        item.find(
+                            'span',
+                            {
+                                'class': 'a-size-base'
+                            }
+                        ).text.strip()
+                    )
+                except ValueError:
+                    logger.error("float error!...")
+
+                saleitem = {
+                    'title': title,
+                    'link': pyshorteners.Shortener(timeout=5).tinyurl.short('https://www.amazon.com' + link),
+                    'saleprice': sale_price,
+                    'oldprice': old_price,
+                    'reviews': reviews
+                }
+                self.deals_list.append(saleitem)
+
+    @staticmethod
+    async def _get_next_page(
+            soup
+    ):
+        pages = soup.find(
+            'span',
+            {
+                'class': 's-pagination-strip'
+            }
+        )
+        if pages:
+            all_pages = pages.find(
+                'ul',
+                {
+                    'class': 'a-unordered-list a-horizontal s-unordered-list-accessibility'
+                }
+            )
+            time.sleep(1)
+            list_page = []
+            if not all_pages:
+                _list = pages.find_all(
+                    "a",
+                    {
+                        "class": "s-pagination-item s-pagination-button"
+                    }
+                )
+                for page in _list:
+                    list_page.append(
+                        'https://www.amazon.com/s?k=' + page.get("href")
+                    )
+                return list_page
+            else:
+                li_urls = all_pages.find_all(
```

Comment:
Use `sync_to_async` method from utils to make this as `async`

Reply:
fixed